src/query.ts

```diff
@@ -201,24 +207,29 @@ class Query {
    * const keyQuery = query.filter('__key__', key);
    * ```
    */
-  filter(property: string, value: {} | null): Query;
-  filter(property: string, operator: Operator, value: {} | null): Query;
+  filter(propertyOrFilter: string | EntityFilter, value?: {} | null): Query;
+  filter(propertyOrFilter: string, operator: Operator, value: {} | null): Query;
   filter(
-    property: string,
-    operatorOrValue: Operator,
+    propertyOrFilter: string | EntityFilter,
+    operatorOrValue?: Operator,
     value?: {} | null
   ): Query {
-    let operator = operatorOrValue as Operator;
-    if (arguments.length === 2) {
-      value = operatorOrValue as {};
-      operator = '=';
-    }
+    if (isFilter(propertyOrFilter)) {
```

Comment:
This is good use of TypeScript's `is` functionality 👍

Reply:
Thanks